sdk-lib/src/main/java/cash/z/ecc/android/sdk/model/Account.kt

```diff
@@ -1,12 +1,113 @@
 package cash.z.ecc.android.sdk.model
 
+import cash.z.ecc.android.sdk.internal.jni.JNI_ACCOUNT_SEED_FP_BYTES_SIZE
+import cash.z.ecc.android.sdk.internal.jni.JNI_ACCOUNT_UUID_BYTES_SIZE
+import cash.z.ecc.android.sdk.internal.model.JniAccount
+
 /**
- * A [ZIP 32](https://zips.z.cash/zip-0032) account index.
+ * Unique identifier for a specific account tracked by a `Synchronizer`.
+ *
+ * Account identifiers are "one-way stable": a given identifier always points to a
+ * specific viewing key within a specific `Synchronizer` instance, but the same viewing
+ * key may have multiple account identifiers over time. In particular, this SDK upholds
+ * the following properties:
+ *
+ * - When an account starts being tracked within a `Synchronizer` instance, a new
+ *   `Account` is generated.
+ * - If an `Account` is present within a `Synchronizer`, it always points to the same
+ *   account.
+ *
+ * What this means is that account identifiers are not stable across "wallet recreation
+ * events". Examples of these include:
+ * - Restoring a wallet from a backed-up seed.
+ * - Importing the same viewing key into two different wallet instances.
  *
- * @param value A 0-based account index.  Must be >= 0.
+ * @param accountUuid The account identifier. Must be length 16.
+ * @param ufvk The account's Unified Full Viewing Key, if any.
+ * @param accountName A human-readable name for the account
+ * @param keySource A string identifier or other metadata describing the source of the seed
+ * @param seedFingerprint The seed fingerprint. Must be length 16.
+ * @param hdAccountIndex ZIP 32 account index
  */
-data class Account(val value: Int) {
+data class Account internal constructor(
+    val accountUuid: ByteArray,
+    val ufvk: String?,
+    val accountName: String?,
+    val keySource: String?,
```

Comment:
keysource we use should be "Keystone" or "Zashi", right?

Reply:
account name probably too 🤔

Reply:
As we discussed with @LukasKorba, I vote for using all caps `KEYSTONE` and `ZASHI` in case the `keySource`. The `accountName` parameter could not be tied by any rules.

Reply:
I used `Keystone` and `Zashi` atm. If we want to set some rules here, we can go either with lowercased or uppcased approach. In iOS world, uppcased is very rare so I would rather use `keystone` and `zashi`.

Reply:
Okay, let's use the lowercase version, then.  @Milan-Cerovsky do you agree?

Reply:
yes lets go with it

Reply:
FYI, I updated the code to follow the lowercase rule